R/workflow.R

```diff
@@ -395,22 +405,18 @@ copy_nonmem_model_from <- function(
   .new_mod[[YAML_MOD_PATH]] <- basename(new_mod_path) # path should be relative to YAML location
 
   # copy control steam to new path
-  if (.update_mod_file) {
-    parent_mod_id <- .parent_mod %>% get_model_path() %>% get_model_id()
-    new_mod_id <- .new_model %>% get_model_id()
-
-    # read parent control stream
-    mod_str <- .parent_mod %>% get_model_path() %>% read_file()
-
-    # replace the $PROBLEM line(s)
-    mod_str <- str_replace(mod_str,
-                "\\$PROB(.|\n)*?\\$",
-                as.character(glue("$PROBLEM {new_mod_id} {.description}\n\n$")))
-
-    # read parent control stream
-    write_file(mod_str, get_model_path(.new_mod))
+  .new_model_path <- .new_mod %>% get_model_path()
+  .parent_model_path <- .parent_mod %>% get_model_path()
+
+  if (fs::file_exists(.new_model_path)) {
+    if (isTRUE(.overwrite_model_file)) {
+      warning(glue("Overwriting file at {.new_model_path} because `.overwrite_model_file=TRUE`"))
+      copy_control_stream(.parent_model_path, .new_model_path, .update_model_file, .description)
+    } else {
+      warning(glue("File already exists at {.new_model_path} -- using existing file because `.overwrite_model_file=FALSE` (default)"))
+    }
   } else {
-    fs::file_copy(get_model_path(.parent_mod), get_model_path(.new_mod))
+    copy_control_stream(.parent_model_path, .new_model_path, .update_model_file, .description)
```

Comment:
instead of nesting this in an else block - a better pattern around control flow is to exit early given an if doesn't work out. Basically the goal is to minimize the need for else statements so people can minimize the number of branches they have to reconcile with. Remind me to discuss this in our next pair programming session

Reply:
makes sense, I'll change it

Reply:
Actually... maybe I'm being dense but I don't see how to do this smoothly without an else block. Since we're not messaging on `.overwrite=TRUE` I changed it to this, which is at least only else instead of two:
```
  if (fs::file_exists(.new_model_path) && !isTRUE(.overwrite)) {
    # if .overwrite != TRUE, warn that file already exists
    warning(glue("File already exists at {.new_model_path} -- using existing file because `.overwrite=FALSE` (default)"))
  } else {
    copy_control_stream(.parent_model_path, .new_model_path, .update_model_file, .description)
  }
```